docs/ReleaseInstructions.md

```diff
@@ -3,6 +3,35 @@
 ### Prior to release: 1 week out
 
 * Ensure the version specified in `docs/config.ini`, `VERSION`, `docs/index.rst`, and `docs/badges.rst` align with the intended release version.
+* Check all dependency version ranges.
+  * Ensure all dependencies are not capped by major version, unless the reason is documented inline.
+    * Example of major version cap: `scikit-learn<2`
+  * Ensure all dependencies have an upper cap, unless the reason is documented inline.
+    * Example of no upper cap: `scikit-learn>=1.0`
```

Comment:
Discussion point: an Introduction of a overly aggressive upper cap would require binary releases every time we want to change the range. This significantly increase work for DLC patching (~1 wk becomes 3-4 weeks).
Pinning pros: repeatable builds
Pinning cons: require more frequent releases as the new CVEs coming.

Reply:
We already have aggressive upper caps on almost every dependency. We are planning to relax some of them this release (Numpy and SciPy)